spec/sim/0006-setspeed-manual-claims/hypothetical_upgrade.scen

```diff
@@ -0,0 +1,103 @@
+
+-- True up balances
+Comptroller ClaimComp DAIWhale
+
+-- Baseline COMP claim speed
+Assert Equal (Comptroller CheckIsComped cDAI) True
+AdvanceBlocks 1000
+Expect Changes (Erc20 Comp TokenBalance DAIWhale) 7.77e2
+Comptroller ClaimComp DAIWhale
+
+-- Propose to apply the patch
+
+From CompHolder (Comp Delegate CompHolder)
+From CompHolder (Governor GovernorAlpha Propose "Disable automatic comp speed refresh and automatic claims and change comp speed" [(Address Unitroller) (Address ComptrollerG7) (Address Unitroller)] [0 0 0] ["_setPendingImplementation(address)" "_become(address)" "_setCompSpeed(address,uint256)"] [[(Address ComptrollerG7)] [(Address Unitroller)] [(Address cUSDC) 30000000000000000]])
+
+-- Vote for, queue, and execute the proposal
+
+MineBlock
+From CompHolder (Governor GovernorAlpha Proposal LastProposal Vote For)
+AdvanceBlocks 20000
+Governor GovernorAlpha Proposal LastProposal Queue
+IncreaseTime 604910
+Governor GovernorAlpha Proposal LastProposal Execute
+ComptrollerImpl ComptrollerG7 MergeABI
+
+-- Check that speed has changed
+Assert Equal (Comptroller CompSpeed cUSDC) 3.0e16
+
+-- Check that DAI is still accruing at the same speed
+Comptroller ClaimComp DAIWhale
+AdvanceBlocks 1000
+Expect Changes (Erc20 Comp TokenBalance DAIWhale) 7.77e2
+Comptroller ClaimComp DAIWhale
+
+-- Check that minting some USDC wont result in any automatic claims
+AdvanceBlocks 1000000
+From USDCWhale (Trx GasPrice 0 (Erc20 USDC Approve cUSDC UInt256Max))
+Expect Changes (Erc20 Comp TokenBalance USDCWhale) 0
+From USDCWhale (Trx GasPrice 0 (CToken cUSDC Mint 10000e6))
+
+-- Prepare second proposal
+From CompHolder (Governor GovernorAlpha Propose "Remove USDC market by setting comp speed to 0" [(Address Unitroller)] [0] ["_setCompSpeed(address,uint256)"] [[(Address cUSDC) 0]])
+
+-- Vote for, queue, and execute the second proposal
+
+MineBlock
+From CompHolder (Governor GovernorAlpha Proposal LastProposal Vote For)
+AdvanceBlocks 20000
+Governor GovernorAlpha Proposal LastProposal Queue
+IncreaseTime 604910
+Governor GovernorAlpha Proposal LastProposal Execute
+
+AdvanceBlocks 1000
+
+-- Check speed is now 0
+Assert Equal (Comptroller CompSpeed cUSDC) 0
+
+-- True up balances
+Comptroller ClaimComp USDCWhale
+
+-- Check no more comp accrued
+AdvanceBlocks 1000000
+Expect Changes (Erc20 Comp TokenBalance USDCWhale) 0
+Comptroller ClaimComp USDCWhale
+
+-- Prepare final proposal
+From CompHolder (Governor GovernorAlpha Propose "Reset USDC comp speed" [(Address Unitroller)] [0] ["_setCompSpeed(address,uint256)"] [[(Address cUSDC) 27000000000000000]])
+
+-- Vote for, queue, and execute the final proposal
+
+MineBlock
+From CompHolder (Governor GovernorAlpha Proposal LastProposal Vote For)
+AdvanceBlocks 20000
+Governor GovernorAlpha Proposal LastProposal Queue
+IncreaseTime 604910
+Governor GovernorAlpha Proposal LastProposal Execute
+
+-- Check new speed
+Assert Equal (Comptroller CompSpeed cUSDC) 2.7e16
```

Comment:
Are there any other checks that Comp is still being distributed appropriately (and ideally other market fns) that you can do?

Reply:
There are now 3 checks:

- USDCWhale accrual changes when the proposal changes USDC flow
- DAIWhale pre & post patch consistency check with supply holdings
- BigBorrower pre & post patch consistency check with both supply and borrow holdings in different tokens

I've added a new file `hypothetical_upgrade_borrower.scen` with the new borrowing check.